dnf-behave-tests/run-tests

```diff
@@ -11,6 +11,16 @@ ENV="$ENV -Ddnf_command=dnf-3"
 
 TAGS=""
 
+OPTIND=1
+while getopts ":p" opt; do
+  case $opt in
+    p)
+      ENV="$ENV -Dpreserve=yes"
+      ;;
```

Comment:
I thing moving shift from line 22 here (without argument) should help.

Reply:
It does, but the `-p` has to be the first argument and if it isn't, the error message is somewhat confusing...

If we're fine with this for the sake of simplicity...

Reply:
True. We can leave getopts out and do the parsing manually:
```
for opt in "$@"; do
  shift
  case $opt in
    -p|--preserve)
      ENV="$ENV -Dpreserve=yes"
      continue
      ;;
  esac
  set -- "$@" "$opt"
done
```

Reply:
Clever! Pushed, thanks.